serve/mlc_serve/engine/staging_engine_worker.py

```diff
@@ -369,8 +369,13 @@ def run_generation_loop_worker(
 
     configure_logging(enable_json_logs, log_level)
     structlog.contextvars.bind_contextvars(**contextvars)
-    model_module = model_module_loader(**model_module_loader_kwargs)
-    worker = GenerationLoopWorker(model_module=model_module)
+
+    try:
+        model_module = model_module_loader(**model_module_loader_kwargs)
+        worker = GenerationLoopWorker(model_module=model_module)
+    except:
+        LOG.exception("An error raised in model initialization.")
```

Comment:
For now we still need to wait for the full timeout at https://github.com/octoml/mlc-llm/blob/batch-serving/serve/mlc_serve/engine/staging_engine.py#L85 even though we hit an error here. Doing `self.ready_event.set()` here is not a good solution.